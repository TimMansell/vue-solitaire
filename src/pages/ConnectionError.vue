<template>
  <GameOverlay center-content show-logo data-test="connection-lost">
    <template #title>No Server Connection</template>
    <template #msg>
      <p>You can wait for the server to come back online or try reconnecting</p>
    </template>
    <template #buttons>
      <ReconnectButton />
    </template>
  </GameOverlay>
</template>

<script>
import { mapGetters } from 'vuex';
import GameOverlay from '@/components/GameOverlay.vue';
import ReconnectButton from '@/components/ReconnectButton.vue';

export default {
  name: 'ConnectionLost',
  components: {
    GameOverlay,
    ReconnectButton,
  },
  computed: {
    ...mapGetters(['hasConnectionError']),
  },
  watch: {
    hasConnectionError(hasConnectionError) {
      if (hasConnectionError) return;

      this.$router.replace('/');
    },
  },
};
</script>
