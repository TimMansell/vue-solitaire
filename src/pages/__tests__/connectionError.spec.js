import { shallowMount } from '@vue/test-utils';
import ConnectionError from '@/pages/ConnectionError.vue';

describe('ConnectionError.vue', () => {
  it('renders the component without crashing', () => {
    const wrapper = shallowMount(ConnectionError);

    expect(wrapper.isVisible()).toBe(true);
  });
});
