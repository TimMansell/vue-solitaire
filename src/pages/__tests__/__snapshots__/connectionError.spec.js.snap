// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConnectionError.vue matches snapshot 1`] = `
<gameoverlay-stub centercontent="true" showlogo="true" data-test="connection-error">No Server Connection<p>You can wait for the server to come back online or try reconnecting</p>
  <reconnectbutton-stub></reconnectbutton-stub>
</gameoverlay-stub>
`;
